### DISCUSSION TITLE
How to force Tailwind to use mobile classes on a desktop device?

### DISCUSSION THREAD

**Discussion by shenders13** [tailwindlabs/tailwindcss#9685] [Help]
I am using Tailwind CSS. My CSS supports two design sizes:

- mobile (<= 768px)
- desktop (> 768px)

For example, I have css like:

`<img class="w-16 md:w-32 src="..." />`

I am building an admin form where the user can fill out form fields (on the left) and then see how it will later be rendered (on the right).

![SYOo8](https://user-images.githubusercontent.com/15011940/198706276-ea5fbf35-07d3-42e2-938d-ab3a0e90a697.jpeg)

The preview on the right has a Desktop/Mobile toggle. I'm unsure how I can force Tailwind to use mobile classes when the user toggles the preview into "Mobile mode".

The problem is that Tailwind classes (e.g. `w-16 md:w-32`) refer to the size of the viewport. However, in this case, I want to render the mobile classes when the user is on a desktop device.

How can I force Tailwind to use mobile classes when the user presses the "Mobile" toggle (even if they are on Desktop device)?

(Cross posted on Stack Overflow [here](https://stackoverflow.com/questions/74227060/how-to-force-tailwind-to-use-mobile-classes-on-a-desktop-device?noredirect=1#comment131050311_74227060))

**Comment by wongjn** [comment]
You could use a class on the container and then have a custom variant or arbitrary variant for desktop view.

For example, mobile view:
```html
<div>
  <img class="w-16 [.desktop_&]:w-32" src="..." />
</div>
```
And then, desktop view:
```html
<div class="desktop">
  <img class="w-16 [.desktop_&]:w-32" src="..." />
</div>
```
If you still want the media query to work too, you could do on mobile:
```html
<div class="mobile">
  <img class="w-16 md:[&:not(.mobile_*)]:w-32 [.desktop_&]:w-32" src="..." />
</div>
```
And for desktop toggle:
```html
<div class="desktop">
  <img class="w-16 md:[&:not(.mobile_*)]:w-32 [.desktop_&]:w-32" src="..." />
</div>
```

**Reply by shenders13** [reply]
Hi @wongjn, thanks for your kind suggestion. 

That would definitely work. But unfortunately, I would have to update the classNames for every element inside the Preview Page (~300 className definitions). I wonder if there is a solution that doesn't leave such a big footprint.

**Comment by adamwathan** [comment]
I like @wongjn's approach! Another idea (and what we do for Tailwind UI for example) is to use an iframe — media queries will report the size of the iframe rather than the size of the viewport.

**Reply by shenders13** [reply]
Hi @adamwathan, good thinking. That will work!

**Reply by shenders13** [reply]
The iframe idea worked well. See below. Thanks @adamwathan!

https://user-images.githubusercontent.com/15011940/210173036-a6c80381-326d-4184-8805-bc8a25c7af45.mov

**Comment by Jakeduncan00** [comment]
If you're not a designer and just want to put together a quick and easy UI for people to use, then Bootstrap might be the way to go. But, if you want complete freedom with your design and the ability to create custom designs and components easily then Tailwind CSS is the better choice  [My Aurora Advocate
](https://www.myadvocateaurora.net/)
